WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -408,6 +408,18 @@ class GutenbergViewController: UIViewController, PostEditor, FeaturedImageDelega
         }
     }
 
+    override func present(_ viewControllerToPresent: UIViewController, animated flag: Bool, completion: (() -> Void)? = nil) {
+        super.present(viewControllerToPresent, animated: flag, completion: completion)
+
+        let presentedView = presentedViewController?.view
+        let presentedViewSwitch = UISwitch.appearance()
+        let presentedViewActionSheet = UIView.appearance(whenContainedInInstancesOf: [UIAlertController.self])
+
+        presentedView?.tintColor = UIColor.editorPrimary
+        presentedViewSwitch.onTintColor = UIColor.editorPrimary
+        presentedViewActionSheet.tintColor = UIColor.editorPrimary
```

Comment:
I believe prefixing the switch and action sheet assignments with `presentedView` is misleading, as I do not believe they are scoped to the presented view. Similar to previous issues referenced in https://github.com/wordpress-mobile/WordPress-iOS/pull/19113#issuecomment-1195965753, this is a global mutation of `UISwitch` and `UIView`/`UIAlertController`, which appears to cause "stale" color issues. 

<details><summary>Stale UISwitch color after opening the editor modal</summary>

https://user-images.githubusercontent.com/438664/181802665-79497f30-2afb-4a6c-bc54-9bcfb7f28069.MP4 

</details>

<details><summary>Stale UIAlertController color after opening the editor modal</summary>

https://user-images.githubusercontent.com/438664/181802688-c68fe3d7-4997-4ae4-998b-dcf34e957a93.MP4

</details>

Ideally we do not have to mutate the global, however, I do not know of an approach for that currently. It is possible a more experienced iOS developer could provide guidance here. 

Is the following an alternative approach for modifying `UISwitch` colors? Could a similar approach be applied here to avoid mutating the global? 

https://github.com/wordpress-mobile/WordPress-iOS/blob/7728b5a8376d7578d265f568f78fb177a03728f1/WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m#L754

A workaround would be reverting the mutation whenever the modal is dismissed. I wonder if we could leverage [`dismiss`](https://developer.apple.com/documentation/uikit/uiviewcontroller/1621505-dismiss) to revert these mutations.

Reply:
@dcalhoun, thank you so much for spotting this! 🙇‍♀️  I was able to figure out a way to [only target switches in the editor](https://github.com/wordpress-mobile/WordPress-iOS/blob/e034bc6b8833a069f49e09bfd63718aeb3ee6965/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L576), by utilising `RCTModalHostViewController`, but wasn't able to follow a similar approach for action sheets.

For action sheets, I came to what feels like a really hacky workaround. Their `tintColor` is now set via the ViewController's [viewWillAppear() function](https://github.com/wordpress-mobile/WordPress-iOS/blob/43b9994672179dc1c0203f46cfce2a36e333bbb5/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L373-L379) and then 'reset' via [the viewWillDisappear function](https://github.com/wordpress-mobile/WordPress-iOS/blob/43b9994672179dc1c0203f46cfce2a36e333bbb5/WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift#L388-L392). I suspect my lack of iOS expertise may be blocking me from seeing a cleaner approach.